src/lib/features/feature-toggle/feature-toggle-utils.ts

```diff
@@ -0,0 +1,279 @@
+                    typeof strategy2.sortOrder === 'number'
+                ) {
+                    return strategy1.sortOrder - strategy2.sortOrder;
+                }
+                return 0;
+            })
+            .map(({ title, sortOrder, ...strategy }) => ({
+                ...strategy,
+                ...(title ? { title } : {}),
+            })),
+    }));
+
+    return result;
+};
+
+export const buildPlaygroundFeaturesFromRows = (
+    rows: any[],
+    dependentFeaturesEnabled: boolean,
+    featureQuery?: IFeatureToggleQuery,
+): FeatureConfigurationClient[] => {
+    let result = rows.reduce((acc, r) => {
+        const feature: PartialDeep<IFeatureToggleClient> = acc[r.name] ?? {
+            strategies: [],
+        };
+        if (isUnseenStrategyRow(feature, r) && !r.strategy_disabled) {
+            feature.strategies?.push(rowToStrategy(r));
+        }
+        if (isNewTag(feature, r)) {
+            addTag(feature, r);
+        }
+        if (featureQuery?.inlineSegmentConstraints && r.segment_id) {
+            addSegmentToStrategy(feature, r);
+        } else if (!featureQuery?.inlineSegmentConstraints && r.segment_id) {
+            addSegmentIdsToStrategy(feature, r);
+        }
+
+        feature.impressionData = r.impression_data;
+        feature.enabled = !!r.enabled;
+        feature.name = r.name;
+        feature.description = r.description;
+        feature.project = r.project;
+        feature.stale = r.stale;
+        feature.type = r.type;
+        feature.lastSeenAt = r.last_seen_at;
+        feature.variants = r.variants || [];
+        feature.project = r.project;
+        feature.lastSeenAt = r.last_seen_at;
+
+        if (r.parent && dependentFeaturesEnabled) {
+            feature.dependencies = feature.dependencies || [];
+            feature.dependencies.push({
+                feature: r.parent,
+                enabled: r.parent_enabled,
+                ...(r.parent_enabled ? { variants: r.parent_variants } : {}),
+            });
+        }
+
+        acc[r.name] = feature;
+        return acc;
+    }, {});
+
+    result = Object.values(result).map(({ strategies, ...rest }) => ({
+        ...rest,
+        strategies: strategies
+            ?.sort((strategy1, strategy2) => {
+                if (
+                    typeof strategy1.sortOrder === 'number' &&
+                    typeof strategy2.sortOrder === 'number'
+                ) {
+                    return strategy1.sortOrder - strategy2.sortOrder;
+                }
+                return 0;
+            })
+            .map(({ title, sortOrder, ...strategy }) => ({
+                ...strategy,
+                ...(title ? { title } : {}),
+            })),
+    }));
+
+    return result;
+};
+
+interface Difference {
+    index: (string | number)[];
+    reason: string;
+    valueA: any;
+    valueB: any;
+}
+
+export function deepDiff(arr1: any[], arr2: any[]): Difference[] | null {
```

Comment:
I don't see this function in the old code. Is it new?

Reply:
does it have tests? what's the advantage over using lodash isEqual?

Reply:
I should have noted that I plan to follow up with some more test coverage in the next PR. The advantage is that it will actually list the two objects that differ, so that you can see what is the difference instead of just getting a boolean value.